languages/python/django-oso/django_oso/partial.py

```diff
@@ -91,17 +91,23 @@ def and_expr(expr: Expression, model: Model, **kwargs):
     return q
 
 
-def compare_expr(expr: Expression, _model: Model, path=(), **kwargs):
+def compare_expr(expr: Expression, model: Model, path=(), **kwargs):
     q = Q()
     (left, right) = expr.args
     left_path = dot_path(left)
     if left_path:
         return COMPARISONS[expr.operator](q, "__".join(path + left_path), right)
     else:
-        if isinstance(right, Model):
+        if isinstance(right, model):
```

Comment:
I don't think this is correct. All the previous code is doing is translating any Django model into its primary key. We pass the authorized model around as `model`, so this change would cause comparisons on subfields that don't inherit from `model` to fail.

Reply:
I should add a test for this.  This branch (from my understanding) only runs for something like `this = post` where `post` is a variable with the `Post` type.  I changed the check because conceivably you could write `this = user`.  This doesn't do any type checking so if the types don't match, we still extract a PK, and then we'd be trying to find a Post that has the PK of the user.

Reply:
I see where you're coming from -- if `left_path` is falsy then there's no dot operation and the LHS must be `_this`, so the only reasonably comparable thing is `model`.

Reply:
Exactly